src/app/branded/step-1/branded-checkout-step-1.spec.js

```diff
@@ -84,6 +84,12 @@ describe('branded checkout step 1', () => {
       $ctrl.initItemConfig()
       expect($ctrl.itemConfig.CAMPAIGN_CODE).toEqual('')
     })
+
+    it('should persist premium-code in item config', () => {
```

Comment:
Can we add a test for no premium code? It would also be good to check that `premiumSelected` is set correctly.